SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_MaterialFarmerTools.cpp

```diff
@@ -568,6 +594,262 @@ void run_from_battles_and_back_to_pokecenter(SingleSwitchProgramEnvironment& env
 
 
 
+
+void move_from_material_farming_to_item_printer(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    env.console.log("Start moving from material farming to item printer.");
+    fly_from_paldea_to_blueberry_entrance(env, context);
+    move_from_blueberry_entrance_to_league_club(env, context);
+    move_from_league_club_entrance_to_item_printer(env, context);
+}
+
+void fly_from_paldea_to_blueberry_entrance(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    int numAttempts = 0;
+    int maxAttempts = 5;
+    bool isFlySuccessful = false;
+
+    while (!isFlySuccessful && numAttempts < maxAttempts){
+        // close all menus
+        pbf_mash_button(context, BUTTON_B, 100);
+
+        numAttempts++;
+
+        open_map_from_overworld(env.program_info(), env.console, context);
+
+        // change from Paldea map to Blueberry map
+        pbf_press_button(context, BUTTON_L, 50, 300);
+
+        // move cursor to bottom right corner
+        pbf_move_left_joystick(context, 255, 255, TICKS_PER_SECOND*5, 50);
+
+        // move cursor to Blueberry academy fast travel point (up-left)
+        pbf_move_left_joystick(context, 0, 0, 76, 50);
```

Comment:
Is this zoom invariant?

Reply:
No. It requires the default zoom level. But this is guaranteed because this routine first closes all menus, then opens the map, which gives us the default zoom level.